Build failure: immich-machine-learning-1.118.2

### Steps To Reproduce

Steps to reproduce the behavior:
1. enable immich server
2. set nixpkgs.config.cudaSupport to true

### Build log

```
error: builder for '/nix/store/306cwbqn0cgqjpya7kjymcgbizzpxrab-immich-machine-learning-1.118.2.drv' failed with exit code 1;
       last 25 log lines:
       > >       assert session.sess_options.inter_op_num_threads == 1
       > E       assert 0 == 1
       > E        +  where 0 = <onnxruntime.capi.onnxruntime_pybind11_state.SessionOptions object at 0x7ffd3da2e330>.inter_op_num_threads
       > E        +    where <onnxruntime.capi.onnxruntime_pybind11_state.SessionOptions object at 0x7ffd3da2e330> = <app.sessions.ort.OrtSession object at 0x7ffd3dfa9d00>.sess_options
       >
       > app/test_main.py:244: AssertionError
       > ------------------------------ Captured log call -------------------------------
       > INFO     ml.log:ort.py:58 Setting execution providers to ['CUDAExecutionProvider', 'CPUExecutionProvider'], in descending order of preference
       > =============================== warnings summary ===============================
       > ../../nix/store/bhj3jdyhr6gpr0rmczb558w4qrkd97fi-python3.12-albumentations-1.4.17/lib/python3.12/site-packages/albumentations/check_version.py:49
       >   /nix/store/bhj3jdyhr6gpr0rmczb558w4qrkd97fi-python3.12-albumentations-1.4.17/lib/python3.12/site-packages/albumentations/check_version.py:49: UserWarning: Error fetching version info <urlopen error [Errno -3] Temporary failure in name resolution>
       >     data = fetch_version_info()
       >
       > app/test_main.py::TestOrtSession::test_sets_default_sess_options_sets_threads_if_non_cpu_and_set_threads
       >   /nix/store/84j57ixnz62k673mnd9ysywlvw6bysfm-python3.12-pydantic-2.8.2/lib/python3.12/site-packages/pydantic/main.py:1059: PydanticDeprecatedSince20: The `__fields__` attribute is deprecated, use `model_fields` instead. Deprecated in Pydantic V2.0 to be removed in V3.0. See Pydantic V2 Migration Guide at https://errors.pydantic.dev/2.8/migration/
       >     warnings.warn(
       >
       > app/test_main.py::TestOrtSession::test_sets_default_sess_options_sets_threads_if_non_cpu_and_set_threads
       >   /nix/store/84j57ixnz62k673mnd9ysywlvw6bysfm-python3.12-pydantic-2.8.2/lib/python3.12/site-packages/pydantic/main.py:1070: PydanticDeprecatedSince20: The `__fields_set__` attribute is deprecated, use `model_fields_set` instead. Deprecated in Pydantic V2.0 to be removed in V3.0. See Pydantic V2 Migration Guide at https://errors.pydantic.dev/2.8/migration/
       >     warnings.warn(
       >
       > -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
       > =========================== short test summary info ============================
       > FAILED app/test_main.py::TestOrtSession::test_sets_default_sess_options - assert 0 == 1
       > ============= 1 failed, 61 passed, 3 skipped, 3 warnings in 2.32s ==============

```

### Additional context

it works without setting cudasupport to true

### Notify maintainers
@jvanbruegge 

### Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/machine-learning/test_main.py
+++ b/machine-learning/test_main.py
@@ -278,8 +278,8 @@ class TestOrtSession:
 
         assert session.provider_options == []
 
-    def test_sets_default_sess_options(self) -> None:
-        session = OrtSession("ViT-B-32__openai")
+    def test_sets_default_sess_options_if_cpu(self) -> None:
+        session = OrtSession("ViT-B-32__openai", providers=["CPUExecutionProvider"])
 
         assert session.sess_options.execution_mode == ort.ExecutionMode.ORT_SEQUENTIAL
         assert session.sess_options.inter_op_num_threads == 1

